x-pack/solutions/security/plugins/security_solution/public/common/store/store.ts

```diff
@@ -66,46 +55,15 @@ export const createStoreFactory = async (
   storage: Storage,
   enableExperimental: ExperimentalFeatures
 ): Promise<Store<State, Action>> => {
-  let signal: { name: string | null; index_mapping_outdated: null | boolean } = {
-    name: null,
-    index_mapping_outdated: null,
-  };
-  try {
-    if (hasAccessToSecuritySolution(coreStart.application.capabilities)) {
-      signal = await coreStart.http.fetch(DETECTION_ENGINE_INDEX_URL, {
-        version: '2023-10-31',
-        method: 'GET',
-      });
-    }
-  } catch {
-    signal = { name: null, index_mapping_outdated: null };
-  }
-
-  const configPatternList = coreStart.uiSettings.get(DEFAULT_INDEX_KEY);
-  let defaultDataView: SourcererModel['defaultDataView'];
-  let kibanaDataViews: SourcererModel['kibanaDataViews'];
-  try {
-    // check for/generate default Security Solution Kibana data view
-    const sourcererDataViews = await createSourcererDataView({
-      body: {
-        patternList: [...configPatternList, ...(signal.name != null ? [signal.name] : [])],
-      },
-      dataViewService: startPlugins.data.dataViews,
-      dataViewId: `${DEFAULT_DATA_VIEW_ID}-${(await startPlugins.spaces?.getActiveSpace())?.id}`,
-    });
-
-    if (sourcererDataViews === undefined) {
-      throw new Error('');
-    }
-    defaultDataView = { ...initDataView, ...sourcererDataViews.defaultDataView };
-    kibanaDataViews = sourcererDataViews.kibanaDataViews.map((dataView: KibanaDataView) => ({
-      ...initDataView,
-      ...dataView,
-    }));
-  } catch (error) {
-    defaultDataView = { ...initDataView, error };
-    kibanaDataViews = [];
-  }
+  const { kibanaDataViews, defaultDataView, signal } = await createDefaultDataView({
+    application: coreStart.application,
+    http: coreStart.http,
+    dataViewService: startPlugins.data.dataViews,
+    uiSettings: coreStart.uiSettings,
+    spaces: startPlugins.spaces,
+    // TODO: (new data view picker) remove this in cleanup phase https://github.com/elastic/security-team/issues/12665
+    skip: enableExperimental.newDataViewPickerEnabled,
```

Comment:
I'm getting a page error when enabling the feature flag. Traced back to the app, drag and drop needs browser fields (LOL)

https://github.com/elastic/kibana/blob/8c811f5e0fec5a233de61dded33603601fe03479/x-pack/solutions/security/plugins/security_solution/public/app/home/index.tsx#L35-L37